src/server/run-websocket-server.ts

```diff
@@ -1,29 +1,54 @@
 import * as WebSocket from "ws";
 import * as Knex from "knex";
+import { EventEmitter } from "events";
 import { JsonRpcRequest } from "../types";
 import { isJsonRpcRequest } from "./is-json-rpc-request";
 import { dispatchJsonRpcRequest } from "./dispatch-json-rpc-request";
 import { makeJsonRpcResponse } from "./make-json-rpc-response";
+import { Subscriptions } from "./subscriptions";
 
 export function runWebsocketServer(db: Knex, port: number): void {
   const websocketServer: WebSocket.Server = new WebSocket.Server({ port });
   websocketServer.on("connection", (websocket: WebSocket): void => {
+    const subscriptions = new Subscriptions();
+
     websocket.on("message", (data: WebSocket.Data): void => {
       try {
         const message = JSON.parse(data as string);
-        if (isJsonRpcRequest(message)) {
+        if (!isJsonRpcRequest(message)) return console.error("bad json rpc message received:", message);
+
+        if (message.method === "subscribe") {
+          const eventName: string = message.params.shift();
+
+          try {
+            const subscription: string = subscriptions.subscribe(eventName, message.params, (data: {}): void => {
```

Comment:
I'm a little unclear what `data` is expected to be here -- is this just the data attached to `emit`, so it can have a pretty arbitrary shape?  Also, what is `message.id` in this context?  Since we don't have a request `id` to match the response with, should it just be `null` for messages pushed to subscribers?

Reply:
1) "data" is just whatever payload we push down the pipe to a subscriber. So in our case it would likely be the database row(s) that we're committing when that event occurs, or some subset of that data.

e.g. for `MarketCreated` I just do `augurEmitter.emit('MarketCreated', dataToInsert);` which corresponds to the row data.

2) Messages pushed to subscribers have no message.id. The immediate response to the subscribe/unsubscribe calls. `message.id` in all the contexts is a Request<->Response mapping which is only valid per connection.